### DISCUSSION TITLE
How to display graphics on screen

### DISCUSSION THREAD

**Discussion by Rochet2** [unikraft/unikraft#364] [Q&A]
I have built and run lvgl that produces an image buffer, but how can I draw the generated image to the screen for KVM/qemu?

In one of the presentation videos I saw an example of lvgl used to draw graphics, but I could not find the example code.
Here is that presentation https://www.youtube.com/watch?v=RMffUqK2XQw&t=1570s

Cannot see any libraries for this in unikraft libs, so currently trying to just access a frame buffer.
Calling `open()` on `/dev/fb` or `/dev/fb0` results in `EACCESS`, so currently trying to see if there is some setting I need in order to enable frame buffer access.

**Comment by vladandrew** [comment]
We might actually need to implement a VirtIO driver for this[1], sadly the presentation was ran on baremetal.

[1] https://lwn.net/Articles/360187/

**Reply by Rochet2** [reply]
I see, thanks for the information.
So I would possibly be able to run on bare metal (linuxu?) and then access frame buffer or something else on the platform to draw image on screen. I will investigate that some more.

**Comment by razvand** [comment]
@Rochet2, since we are already (viciously) using [the Unikraft Discord server](https://bit.ly/UnikraftDiscord) for discussions, meetings, brainstorming and jokes (mostly bad, we do what we can), we will close the Discussions tab. We will do this on Friday, March 31, 2023.

Please move any discussions or questions on [the Unikraft Discord server](https://bit.ly/UnikraftDiscord). See you all there! :handshake:
